src/powerset.rs

```diff
@@ -58,6 +58,17 @@ impl<I> Iterator for Powerset<I>
         }
     }
 
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let k = self.combs.k();
+        // Total bounds for source iterator.
+        let (n_min, n_max) = self.combs.src().size_hint();
+        // Total bounds for the current combinations.
+        let (mut low, mut upp) = self.combs.size_hint();
+        low = remaining_for(low, n_min, k).unwrap_or(usize::MAX);
+        upp = upp.and_then(|upp| remaining_for(upp, n_max?, k));
```

Comment:
It *feels* as if this replicates the logic of `size_hint::add` - what do you think @Philippe-Cholet? If so, can we somehow use `size_hint::add`?